fastlane/Fastfile

```diff
@@ -147,6 +140,49 @@ lane :update_hybrid_common do |options|
   update_ios_pod(options)
 end
 
+desc "Generate docs"
+lane :generate_docs do
+  version_number = current_version_number
+  docs_repo_base_url = "https://github.com/RevenueCat/"
+  docs_repo_name = "react-native-purchases-docs"
+  docs_repo_url = File.join(docs_repo_base_url, docs_repo_name)
+  hosting_base_path = File.join(docs_repo_name, version_number)
+  
+  Dir.mktmpdir do |docs_generation_folder|
+    # Must be run from the root dir
+    Dir.chdir("..") do
+      sh(
+        "npx",
+        "typedoc",
+        "--out",
+        docs_generation_folder
+      )
+      
+      docs_index_path = File.join(Dir.pwd, "scripts/docs/index.html")
+
+      # clone docs repo
+      Dir.mktmpdir do |docs_repo_clone_dir|
+        Dir.chdir(docs_repo_clone_dir) do
+          sh("git", "clone", docs_repo_url)
+          Dir.chdir(docs_repo_name) do
+            # copy docs generated in the previous step into the docs folder and push the changes
+            docs_destination_folder = "docs/#{version_number}"
+            index_destination_path = "docs/index.html"
+            FileUtils.cp_r docs_generation_folder, docs_destination_folder
```

Comment:
Should we fail if destination folder already exists? Or just allow a silent overwrite? I guess I'm ok with an overwrite but it might not delete old files if it needs to remove any, so maybe we want to delete the old folder first?

Reply:
That's a good point. Thinking about this now maybe we should move this code into the shared plugin because it's sort of duplicated in 3 places now, so it would be nice to get that fix across the board.